### PR TITLE
[CI] Run only accelerator PT2E tests on GPU

### PR DESCRIPTION
**Pull request by iamzainhuda**
## Summary

- keep the original PT2E test files CPU-only
- move the eight accelerator variants into three dedicated `*_with_gpu.py` files
- run those files through one shared CI script so CUDA and ROCm execute the same coverage
- report the 25 slowest test durations for future profiling

The four-worker CPU PT2E workflow continues to run the complete `test/quantization/pt2e` suite. This removes redundant CPU-oriented PT2E execution from expensive GPU runners without removing tests from CI.

## CI results

Compared [latest main](https://github.com/pytorch/ao/actions/runs/32980155475) with the [latest PR run](https://github.com/pytorch/ao/actions/runs/33096887298) on the same runner classes and matrix configurations:

| Environment | Metric | Before | After | Reduction |
| --- | --- | ---: | ---: | ---: |
| CUDA Nightly | pytest | 49:00 | [6:22](https://github.com/pytorch/ao/actions/runs/33096887298/job/98604023355) | 87.0% |
| CUDA Nightly | full job | 1:04:28 | 0:20:35 | 68.1% |
| CUDA 2.13 | pytest | 46:00 | [5:49](https://github.com/pytorch/ao/actions/runs/33096887298/job/98604023616) | 87.4% |
| CUDA 2.13 | full job | 1:02:31 | 0:20:27 | 67.3% |
| ROCm Nightly | pytest | 29:28 | [3:49](https://github.com/pytorch/ao/actions/runs/33096887298/job/98604023605) | 87.0% |
| ROCm Nightly | full job | 0:43:40 | 0:11:52 | 72.8% |

All three jobs collected and passed the eight accelerator tests:

- CUDA Nightly: `8 passed, 8 subtests passed` in `382.41s (0:06:22)`
- CUDA 2.13: `8 passed, 8 subtests passed` in `349.13s (0:05:49)`
- ROCm Nightly: `8 passed, 8 subtests passed` in `229.38s (0:03:49)`

Both complete PT2E CPU matrix jobs also passed after the file split.

## Coverage audit

The dedicated workflow exists for accelerator PT2E coverage but previously ran every non-x86 PT2E test. The accelerator variants are now explicit in:

- `test_quantize_pt2e_qat_with_gpu.py`: four conv-bn fusion tests
- `test_quantize_pt2e_with_gpu.py`: accelerator batch-norm and train/eval tests
- `test_learnable_fake_quantize_with_gpu.py`: device compatibility and numerical consistency tests

Shared helpers keep the CPU and accelerator variants on the same test logic. No same-GPU xdist is introduced, avoiding GPU-memory contention and distributed-test collisions.

## Test plan

- `bash .github/scripts/ci_test_pt2e_gpu.sh --collect-only -q` (8 tests collected)
- targeted original-file CPU variants (`8 passed`)
- accelerator files without a GPU (`8 skipped`)
- local CUDA accelerator run (`8 passed`)
- full PT2E collection (`617 tests collected`)
- `bash -n .github/scripts/ci_test_pt2e_gpu.sh`
- Ruff check and format
- `git diff --check`
- CUDA Nightly dedicated job: passed
- CUDA 2.13 dedicated job: passed
- ROCm Nightly dedicated job: passed
- CPU Nightly complete PT2E job: passed
- CPU 2.13 complete PT2E job: passed

### CODE REVIEW

File: .github/scripts/ci_test_pt2e_gpu.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# The CPU PT2E workflow owns the full suite. Keep this list limited to tests
+# that exercise accelerator-specific paths.
+pt2e_gpu_tests=(
```
Comment (vkuzo) [inline .github/scripts/ci_test_pt2e_gpu.sh:6]: I think this will be confusing for people who write tests

How about this instead:
* make existing test files cpu-only
* refactor the tests that need GPU to new test files
* run the new test files from the CI job

For example

```
# old

# test
test/quantization/pt2e/test_quantize_pt2e_qat.py
# run test - same on cpu and gpu
pytest test/quantization/pt2e/test_quantize_pt2e_qat.py

# new

# test files
test/quantization/pt2e/test_quantize_pt2e_qat.py
test/quantization/pt2e/test_quantize_pt2e_qat_with_gpu.py

# run on cpu
pytest test/quantization/pt2e/test_quantize_pt2e_qat.py
pytest test/quantization/pt2e/test_quantize_pt2e_qat_with_gpu.py

# run on gpu
pytest test/quantization/pt2e/test_quantize_pt2e_qat_with_gpu.py
```

Comment (iamzainhuda) [reply .github/scripts/ci_test_pt2e_gpu.sh:6]: Implemented this in d9b9c4219. The existing files now own CPU-only variants, the accelerator variants live in three dedicated `*_with_gpu.py` files, and the GPU script runs those files directly. Shared helpers keep both variants on the same test logic. The fresh CUDA Nightly, CUDA 2.13, ROCm, CPU Nightly, and CPU 2.13 jobs all pass; I updated the PR body with the new timings.

Comment (vkuzo) [reply .github/scripts/ci_test_pt2e_gpu.sh:6]: great, let's remove the bash script and just run it from the .yml file

**Comment by vkuzo** [comment]
@claude review
